unit-tests/src/test/java/com/novoda/merlin/MerlinsBeardShould.java

```diff
@@ -53,4 +54,27 @@ public void returnTrueForIsConnectedWhenNetworkConnectionIsAvailable() {
         assertThat(merlinsBeard.isConnected()).isTrue();
     }
 
-}
\ No newline at end of file
+    @Test
+    public void returnTrueForIsConnectedToWifiWhenNetworkConnectedToWifiIsConnected() {
+        when(mockConnectivityManager.getNetworkInfo(WIFI.getValue())).thenReturn(mockNetworkInfo);
+        when(mockNetworkInfo.isConnected()).thenReturn(true);
+
+        assertThat(merlinsBeard.isConnectedTo(WIFI)).isTrue();
+    }
+
+    @Test
+    public void returnFalseForIsConnectedToWifiWhenNetworkConnectedToWifiIsNotConnected() {
```

Comment:
An implementation of isConnectedToWifi like getActiveNetworkInfo != null &&
!= mobile and connected would probably pass..
Am 29.06.2015 14:36 schrieb "Ferran Garriga" <notifications@github.com>:

> In unit-tests/src/test/java/com/novoda/merlin/MerlinsBeardShould.java
> <https://github.com/novoda/merlin/pull/58#discussion_r33458770>:
>
> > \ No newline at end of file
> > +    @Test
> > +    public void returnTrueForIsConnectedToWifiWhenNetworkConnectedToWifiIsConnected() {
> > +        when(mockConnectivityManager.getNetworkInfo(WIFI.getValue())).thenReturn(mockNetworkInfo);
> > +        when(mockNetworkInfo.isConnected()).thenReturn(true);
> > +
> > +        assertThat(merlinsBeard.isConnectedTo(WIFI)).isTrue();
> > +    }
> > +
> > +    @Test
> > +    public void returnFalseForIsConnectedToWifiWhenNetworkConnectedToWifiIsNotConnected() {
>
> I think that your case is already covered. getActiveNetworkInfo() is not
> invoked when checking for a specific network type availability.
>
> When is connected and wifi is false might be this case
> returnFalseForIsConnectedToWifiWhenNetworkConnectionIsNotAvailable there
> it could be connected but the active wifi interface is null (returned as
> so by the Android framework)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/merlin/pull/58/files#r33458770>.
>